File: WebApplication/src/suppliers/dtos/search/supplier-group/supplier-root-group.dto.ts
import { BaseSupplierGroupDto } from "./base/base-supplier-group.dto";
import { SupplierSubordinateGroupDto } from "./supplier-subordinate-group.dto";

export class SupplierRootGroupDto extends BaseSupplierGroupDto {
    supplierSubordinateGroup: SupplierSubordinateGroupDto[] = [];
}

File: Sc.Models/Dtos/Suppliers/Search/SupplierGroup/SupplierRootGroupDto.cs
using Sc.Models.Dtos.Suppliers.Search.SupplierGroup.Base;

namespace Sc.Models.Dtos.Suppliers.Search.SupplierGroup
{
    public class SupplierRootGroupDto : BaseSupplierGroupDto
    {
        public List<SupplierSubordinateGroupDto> SupplierSubordinateGroup { get; set; } = new List<SupplierSubordinateGroupDto>();
    }
}
